frontend/packages/ceph-storage-plugin/src/components/modals/restore-pvc-modal/restore-pvc-modal.tsx

```diff
@@ -0,0 +1,145 @@
+import './_restore-pvc-modal.scss';
+
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Form, FormGroup, Grid, GridItem, TextInput } from '@patternfly/react-core';
+import {
+  HandlePromiseProps,
+  ResourceIcon,
+  withHandlePromise,
+} from '@console/internal/components/utils/index';
+import { K8sResourceKind, k8sCreate } from '@console/internal/module/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '@console/internal/components/factory';
+import { PersistentVolumeClaimModel } from '@console/internal/models';
+
+export const RestorePVCModal = withHandlePromise((props: RestorePVCModalProps) => {
+  const { close, cancel, resource, errorMessage, inProgress } = props;
+  const [buttonDisabled, setButton] = React.useState(false);
+
+  const [restorePVCName, setPVCName] = React.useState(
+    `${_.get(resource, 'metadata.name')}-restore`,
+  );
+  const onValueChange = (value: string) => {
+    if (value === '') {
+      setButton(true);
+    } else {
+      setButton(false);
+    }
+    setPVCName(value);
+  };
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    const restorePVCTemplate: K8sResourceKind = {
+      apiVersion: 'v1',
+      kind: 'PersistentVolumeClaim',
+      metadata: {
+        name: `${restorePVCName}`,
+      },
+      spec: {
+        storageClassName: 'csi-hostpath-sc', // to be checked
+        dataSource: {
+          name: `${_.get(resource, 'metadata.name')}`,
+          kind: 'VolumeSnapshot',
+          apiGroup: 'snapshot.storage.k8s.io/v1apha1',
```

Comment:
use model to define it. https://github.com/openshift/console/pull/3976/files#diff-fba3e1fe6074a74d7fd9a0370ecbdc08R31

Reply:
ack